src/support/slack/commands/onboard.js

```diff
@@ -67,43 +60,75 @@ function OnboardCommand(context) {
     const { DEFAULT_ORGANIZATION_ID: defaultOrgId } = context.env;
 
     try {
-      const [baseURLInput] = args;
-
+      const [baseURLInput, imsOrgID, profileName = 'default'] = args;
       const baseURL = extractURLFromSlackInput(baseURLInput);
 
-      if (!baseURL) {
+      await say(`:gear: Applying ${profileName} profile.`);
+
+      if (!isValidUrl(baseURL)) {
         await say(':warning: Please provide a valid site base URL.');
         return;
       }
 
-      // see if the site was added previously
-      let site = await Site.findByBaseURL(baseURL);
+      if (!OrganizationModel.IMS_ORG_ID_REGEX.test(imsOrgID)) {
+        await say(':warning: Please provide a valid IMS Org ID.');
+        return;
+      }
+
+      // check if the organization with IMS Org ID already exists; create if it doesn't
+      let organization = await Organization.findByImsOrgId(imsOrgID);
+      if (!organization) {
+        organization = await Organization.create(context);
+        const message = `:white_check_mark: A new organization has been created. Organization ID: ${organization.id} Organization name: ${organization.name} IMS Org ID: ${imsOrgID}.`;
+        await say(message);
+        log.info(message);
+      }
 
-      // if not, add the site to the star catalogue
+      // check if the site already exists; create if it doesn't
+      let site = await Site.findByBaseURL(baseURL);
       if (!site) {
         const deliveryType = await findDeliveryType(baseURL);
-        const isLive = true;
+        const isLive = deliveryType === SiteModel.DELIVERY_TYPES.AEM_EDGE;
 
         site = await Site.create({
           baseURL, deliveryType, isLive, organizationId: defaultOrgId,
         });
       }
 
+      const profile = await loadProfileConfig(profileName);
```

Comment:
if within `loadProfileConfig` an exception occurs, it is caught and the method returns `undefined`. this will lead to issues below, e.g. `profile.audits`. please make more robust. minimally you could check `if (!isObject(profile?.audits)) { say/log }` or similar. same for imports